frontend/public/components/modals/cluster-channel-modal.tsx

```diff
@@ -13,85 +13,59 @@ import {
+  const { t } = useTranslation();
+  const availableChannels = getAvailableClusterChannels(cv).reduce((o, val) => {
+    o[val] = val;
+    return o;
+  }, {});
+  const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
     e.preventDefault();
-    const { cv } = this.props;
-    const { selectedChannel } = this.state;
-    const patch = [{ op: 'add', path: '/spec/channel', value: selectedChannel }];
-    this.handlePromise(k8sPatch(ClusterVersionModel, cv, patch)).then(this.props.close);
+    const patch = [{ op: 'add', path: '/spec/channel', value: channel }];
+    return handlePromise(k8sPatch(ClusterVersionModel, cv, patch), close);
   };
 
-  _cancel = () => {
-    this.props.close();
-  };
-
-  _change = (selectedChannel) => {
-    this.setState({ selectedChannel });
-  };
-
-  render() {
-    const { cv } = this.props;
-    const availableChannels = getAvailableClusterChannels(cv).reduce((o, val) => {
-      o[val] = val;
-      return o;
-    }, {});
-    return (
-      <form
-        onSubmit={this._submit}
-        name="form"
-        className="modal-content modal-content--no-inner-scroll"
-      >
-        <ModalTitle>Update Channel</ModalTitle>
-        <ModalBody>
-          <p>
-            Select a channel that reflects your desired version. Critical security updates will be
-            delivered to any vulnerable channels.
-          </p>
-          <p>
-            <ChannelDocLink />
-          </p>
-          <div className="form-group">
-            <label htmlFor="channel_dropdown">Select Channel</label>
-            <Dropdown
-              className="cluster-channel-modal__dropdown"
-              id="channel_dropdown"
-              items={availableChannels}
-              onChange={this._change}
-              selectedKey={cv.spec.channel}
-              title="Select Channel"
-            />
-          </div>
-        </ModalBody>
-        <ModalSubmitFooter
-          errorMessage={this.state.errorMessage}
-          inProgress={this.state.inProgress}
-          submitText="Save"
-          cancel={this._cancel}
-        />
-      </form>
-    );
-  }
-}
+  return (
+    <form onSubmit={submit} name="form" className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>{t('cluster-channel-modal~Update channel')}</ModalTitle>
+      <ModalBody>
+        <p>
+          {t(
+            'cluster-channel-modal~Select a channel that reflects your desired version. Critical security updates will be delivered to any vulnerable channels.',
+          )}
+        </p>
+        <p>
+          <ChannelDocLink />
+        </p>
+        <div className="form-group">
+          <label htmlFor="channel_dropdown">{t('cluster-channel-modal~Select channel')}</label>
+          <Dropdown
+            className="cluster-channel-modal__dropdown"
+            id="channel_dropdown"
+            items={availableChannels}
+            onChange={(newChannel: string) => setChannel(newChannel)}
```

Comment:
```suggestion
            onChange={setChannel}
```

Reply:
Suggested change results in the following error:
`Warning: State updates from the useState() and useReducer() Hooks don't support the second callback argument. To execute a side effect after rendering, declare it in the component body with useEffect().`

I ran in to this during development and you said I needed to use the arrow function in order to remove the second parameter.